src/components/ReplaceServerDetails.js

```diff
@@ -0,0 +1,321 @@
+// (c) Copyright 2017-2018 SUSE LLC
+/**
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+import React, { Component } from 'react';
+import { translate } from '../localization/localize.js';
+import { ActionButton } from '../components/Buttons.js';
+import { InputLine } from '../components/InputLine.js';
+import { ListDropdown } from '../components/ListDropdown.js';
+import { IpV4AddressValidator, MacAddressValidator } from '../utils/InputValidators.js';
+import { INPUT_STATUS } from '../utils/constants.js';
+import { genUID, getAvailableServerIds, maskPassword } from '../utils/ModelUtils.js';
+import HelpText from '../components/HelpText.js';
+import { Map, fromJS } from 'immutable';
+import {fetchJson} from "../utils/RestUtils";
+
+class ReplaceServerDetails extends Component {
+  constructor(props) {
+    super(props);
+
+    this.allInputsStatus = {
```

Comment:
These status fields should be part of state, since they directly affect what is being rendered

Reply:
will see what I can do